### DISCUSSION TITLE
Checking the missing uppercase in the beginning of the translations

### DISCUSSION THREAD

**Discussion by amieiro** [GlotPress/GlotPress-WP#1441] [Ideas]
I want to add a validation to check for the missing uppercase in the beginning of the translations. This checking only makes sense for the Latin alphabet locales. I have in my mind 2 different approaches to do this:

1. Make an array with the Latin alphabet languages and use it on the warning check. This is the easy way, but I think it is not the good one.
2. Add a new `alphabet` property to the [GP_Locale](https://github.com/GlotPress/GlotPress/blob/develop/locales/locales.php#L4) class, and then add the specific value in each locale, in the [GP_Locales](https://github.com/GlotPress/GlotPress/blob/develop/locales/locales.php#L117) class. This is a longer work, but I think this is the good approach.

We have information around the alphabet used in each locale in a few places:
* [List of languages by writing system](https://en.wikipedia.org/wiki/List_of_languages_by_writing_system)
* [List of Latin-script alphabets](https://en.wikipedia.org/wiki/List_of_Latin-script_alphabets)
* [Appendix:Latin script/alphabets](https://en.wiktionary.org/wiki/Appendix:Latin_script/alphabets)

**Do you have some comments or feedback? Please, write them in the discussion.**

**Comment by psmits1567** [comment]
You probably also need to add a check for the first position not being a blank. A lot of times the strings containing a lowercase first char, start with a blank before.

**Reply by amieiro** [reply]
I think I will do a trim with both strings, to remove blank spaces at the begging and at the end, to avoid these problems. The warnings related with the empty spaces were addressed in the PR https://github.com/GlotPress/GlotPress/pull/1427. Thank you for the feedback.

**Comment by tobifjellner** [comment]
There may be situations where it's correct to start the translation with a lowercase letter even if the source strings starts with uppercase. In Swedish, that could be in bullet lists, or if some strings get combined in the UI.
Still, since a warning can be discarded, this sounds like a good idea.

**Reply by amieiro** [reply]
Thank you for the feedback. It will be a warning, so it could be discarded.

**Comment by pedro-mendonca** [comment]
One of those situations is the set of tags used in core blocks, where the originals are in lowercase, and are probably many others.
I can't imagine this working in anyway different of the current warnings that the editor must discard for wrong URLs and placeolders.

**Reply by amieiro** [reply]
Yes, it will be a warning, so it could be discarded. Thank you for the feedback.

**Comment by NekoJonez** [comment]
Personally, the way it works in GlotDict is fantastic. Yet, it's sometimes "annoying" when a certain locale has to switch up the string a bit making a variable in front so a lowercase is where an uppercase should be.

**Reply by amieiro** [reply]
Yes, in some cases we will have situations like this, but the warning could be discarded. Thank you for the feedback.

**Comment by naokomc** [comment]
I'd love to see this change implemented to GlotPress 👍 

One thing I noticed is the warning sometimes confuses contributors. Even though it's a dismissable warning, some people really hesitate to ignore it and don't know what to do for an edge case that's a false-positive.

Currently GlotDict says:

> Warning: The translation is missing an initial uppercase letter as "A" is an initial uppercase letter present on the original string.

I think it's more friendly if we can say something more explicit:

> Warning: The translation starts with an uppercase, unlike the original. If it was intentional, you can discard this warning.

**Reply by amieiro** [reply]
Good point, Naoko. I will use it in the PR. Thank you for the feedback.

**Comment by amieiro** [comment]
Implemented here https://github.com/GlotPress/GlotPress/pull/1450.
